src/k8s/pkg/k8sd/database/feature_status.go

```diff
@@ -0,0 +1,79 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/microcluster/cluster"
+)
+
+var featureStatusStmts = struct {
+	select_ int
+	upsert_ int
+}{
+	select_: MustPrepareStatement("feature-status", "select.sql"),
+	upsert_: MustPrepareStatement("feature-status", "upsert.sql"),
+}
```

Comment:
let's to a `map[string]int` like in other places, to avoid the duplicate lines here

Reply:
tbh I really like every bit of type safety that I can get :D